flybywiresim/a32nx
# feat(efb): EFB Landing Performance ⚡

## Summary of Changes
Adds a landing performance page to the Performance section of the EFB. Previously raised in a PR with takeoff performance, but TO perf needed more work so merging seperately to keep the level of changes that needs reviewing small.

## Screenshots (if necessary)
![image](https://user-images.githubusercontent.com/15931560/108629316-2b28c100-7457-11eb-8154-0dc32e55f2ae.png)
![image](https://user-images.githubusercontent.com/15931560/108629333-40055480-7457-11eb-8ea1-ee99abc95bb6.png)
![image](https://user-images.githubusercontent.com/15931560/108629383-907cb200-7457-11eb-8ffd-dcd442a434f6.png)
![image](https://user-images.githubusercontent.com/15931560/108629375-8490f000-7457-11eb-97a6-6c92ec005373.png)
![image](https://user-images.githubusercontent.com/15931560/108629307-2401b300-7457-11eb-8be8-960a27fad3c2.png)


## References
Quick Reference Handbook, In-Flight Performance, Landing Distance Without Failure (Applicable to: D-AVVI, D-AXAJ, F-WWTQ, PR-OBJ)
![image](https://user-images.githubusercontent.com/15931560/108266196-b78a6980-7161-11eb-8a30-07043fde8bb7.png)
![image](https://user-images.githubusercontent.com/15931560/108266236-c2dd9500-7161-11eb-87f5-b4db3223d95f.png)
![image](https://user-images.githubusercontent.com/15931560/108266361-ed2f5280-7161-11eb-88a0-932f967bce07.png)

15% safety margin is determined from the recommended safety margin in the QRH:
![image](https://user-images.githubusercontent.com/15931560/108266758-662eaa00-7162-11eb-8215-42594a44b5b5.png)



## Additional context
Discord username: TheMightyMat#0585

## Testing instructions
* Ensure landing distances vary as expected with changes in input values.
* Ensure landing distances on runway visualisation match output data
* Ensure distances that are longer than LDA are highlighted in red
* Test for display issues for extreme values and realistic values
* Ensure inputs are limited to sensible values

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link
